xbmc/platform/darwin/osx/network/NetworkOsx.cpp

```diff
@@ -192,11 +211,11 @@ std::vector<std::string> CNetworkOsx::GetNameServers()
     pollFd.events = POLLIN;
     pollFd.revents = 0;
 
-    int pollResult = poll(&pollFd, 1, 1000); // Poll for 1 second
+    int pollResult = poll(&pollFd, 1, 100); // Wait for 100 milliseconds
     if (pollResult > 0 && (pollFd.revents & POLLIN))
     {
-      char buffer[256] = {'\0'};
-      if (fgets(buffer, sizeof(buffer), pipe))
+      std::string buffer(256, '\n');
```

Comment:
Why is it better to fill with line breaks rather than zeros?

Reply:
It was easy because of the split. Otherwise, if you have a single DNS entry you'll end up with a second string (with the remaining buffer size filled with zeros).
Regardless, I agree this is hard to grasp and a bit hacky. Changed to zeros again but that requires that the string is resized until the first null char is found (so that we check for emptyness).